<launch>
  <arg name="usb_port"                default="/dev/ttyUSB_dynamixel"/>
  <arg name="dxl_baud_rate"           default="1000000"/>
  <arg name="namespace"               default="manipulator"/>

  <arg name="use_moveit"              default="true"/>
  <arg name="use_joint_state"         default="true"/>
  <arg name="use_cmd_vel"             default="false"/>

  <param name="dynamixel_info"          value="$(find aerial_manipulators_control)/config/asap_manipulator_4r_controllers.yaml"/>

  <node name="$(arg namespace)" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers"
        required="true" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <param name="use_moveit"              value="$(arg use_moveit)"/>
    <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
    <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
    <rosparam>
      publish_period: 0.010
      dxl_read_period: 0.010
      dxl_write_period: 0.010
      mobile_robot_config:                <!--this values will be set when 'use_cmd_vel' is true-->
        seperation_between_wheels: 0.160  <!--default value is set by reference of TB3-->
        radius_of_wheel: 0.033            <!--default value is set by reference of TB3-->
    </rosparam>
  </node>

  <node name="manipulator_control_node" pkg="aerial_manipulators_control" type="manipulator_control_node" output="screen" ns="$(arg namespace)">
        <param name="rate" value="30"/>
        <param name="robot_name" value="asap_manipulator_4r"/>
        <param name="joint_group_name" value="asap_manipulator_4r_arm"/>
        <param name="parameters_file" value="/config/asap_manipulator_4r_parameters.yaml"/>
  </node>

</launch>
